### DISCUSSION TITLE
Is it possible to references env variables inside env properties?

### DISCUSSION THREAD

**Discussion by mochadwi** [community/community#26241] [Actions]
<pre><code class="lang-auto">env:
  FLAVOUR: ${{ github.base_ref }}
  TYPE: debug
  BUILD_VARIANT: $FLAVOUR$TYPE
</code></pre>
<p>When trying to echo the <code>$BUILD_VARIANT</code> it prints literal <code>$FLAVOUR$TYPE</code> instead. Using double-quoted string also not works.</p>
<p>Is there any workaround suggested? or is it not possible atm?</p>

**Comment by mochadwi** [comment, marked as answer]
<p>When debugging, this seems not possible atm, therefore we use hardcoded string instead or manually <code>export BUILD_VARIANT="${FLAVOUR}${TYPE}"</code> this ways instead</p>
<p>As shown in the below images, it’s showing literal <code>$FLAVOUR</code> instead of the value <code>dev</code></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/7/2/7298c6ae51f39a152f91ef2fa46c3d0320182f78.png" data-download-href="/uploads/short-url/glLKGbBKl8j5qRAmGhPcFeC2XUI.png?dl=1" title="Screen Shot 2022-06-23 at 1.55.52 PM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/7057959/181096378-e6489e0b-a1ef-4468-9778-6697abcc55b4.png" alt="Screen Shot 2022-06-23 at 1.55.52 PM" data-base62-sha1="glLKGbBKl8j5qRAmGhPcFeC2XUI" width="689" height="312" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/7/2/7298c6ae51f39a152f91ef2fa46c3d0320182f78_2_689x312.png, https://aws1.discourse-cdn.com/github/optimized/3X/7/2/7298c6ae51f39a152f91ef2fa46c3d0320182f78_2_1033x468.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/7/2/7298c6ae51f39a152f91ef2fa46c3d0320182f78.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/2/7298c6ae51f39a152f91ef2fa46c3d0320182f78_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">Screen Shot 2022-06-23 at 1.55.52 PM</span><span class="informations">1194×540 50.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>This is not works as well:</p>
<pre><code class="lang-auto">BUILD_VARIANT: ${{ FLAVOUR }}
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/a/2a97a49e688ec50cdb94ad81699923d96039b590.png" data-download-href="/uploads/short-url/64MWK6Ao2bChs08edXKPiHyvri0.png?dl=1" title="Screen Shot 2022-06-23 at 1.59.31 PM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/7057959/181096379-ebb1d28c-7162-439f-9a6f-0a62cc4ca898.png" alt="Screen Shot 2022-06-23 at 1.59.31 PM" data-base62-sha1="64MWK6Ao2bChs08edXKPiHyvri0" width="690" height="184" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/a/2a97a49e688ec50cdb94ad81699923d96039b590_2_690x184.png, https://aws1.discourse-cdn.com/github/optimized/3X/2/a/2a97a49e688ec50cdb94ad81699923d96039b590_2_1035x276.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/2/a/2a97a49e688ec50cdb94ad81699923d96039b590_2_1380x368.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/a/2a97a49e688ec50cdb94ad81699923d96039b590_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">Screen Shot 2022-06-23 at 1.59.31 PM</span><span class="informations">2044×548 85 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>–</p>
<pre><code class="lang-auto">BUILD_VARIANT: ${{ env.FLAVOUR }}
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/7/c705d926917ca5f759da5fa2fda87bc225b4fb51.png" data-download-href="/uploads/short-url/soDxSsBDCeWagtGoMo198auH8Pf.png?dl=1" title="Screen Shot 2022-06-23 at 2.00.18 PM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/7057959/181096380-f90adfbd-76c2-470e-ade0-08c2949ec6af.png" alt="Screen Shot 2022-06-23 at 2.00.18 PM" data-base62-sha1="soDxSsBDCeWagtGoMo198auH8Pf" width="690" height="180" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/7/c705d926917ca5f759da5fa2fda87bc225b4fb51_2_690x180.png, https://aws1.discourse-cdn.com/github/optimized/3X/c/7/c705d926917ca5f759da5fa2fda87bc225b4fb51_2_1035x270.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/c/7/c705d926917ca5f759da5fa2fda87bc225b4fb51_2_1380x360.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/7/c705d926917ca5f759da5fa2fda87bc225b4fb51_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">Screen Shot 2022-06-23 at 2.00.18 PM</span><span class="informations">2040×534 74.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>–</p>
<pre><code class="lang-auto">BUILD_VARIANT: ${{ $FLAVOUR }}
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/b/4bd929943127632b711749518f6f570a5c85f365.png" data-download-href="/uploads/short-url/aOZ4ScIbcGOOPDdEnb9jkp4kEfz.png?dl=1" title="Screen Shot 2022-06-23 at 1.59.43 PM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/7057959/181096381-d22a77e3-0e76-4e17-9c77-e06afb3b0c9a.png" alt="Screen Shot 2022-06-23 at 1.59.43 PM" data-base62-sha1="aOZ4ScIbcGOOPDdEnb9jkp4kEfz" width="690" height="176" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/b/4bd929943127632b711749518f6f570a5c85f365_2_690x176.png, https://aws1.discourse-cdn.com/github/optimized/3X/4/b/4bd929943127632b711749518f6f570a5c85f365_2_1035x264.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/4/b/4bd929943127632b711749518f6f570a5c85f365_2_1380x352.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/b/4bd929943127632b711749518f6f570a5c85f365_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">Screen Shot 2022-06-23 at 1.59.43 PM</span><span class="informations">2030×518 76.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by jsoref** [comment]
<p>Instead of using <code>env:</code>, you can append something to <code>$GITHUB_ENV</code> in the preceding step.</p>
<p>If you don’t want it to persist, the strip that uses it can remove the line from that file (I’d use <code>perl -ni -e 'next if /^BUILDVARIANT=/; print' "$GITHUB_ENV"</code>, but you can manage it in many ways.</p>
